sdk/devcenter/Azure.Developer.DevCenter/tests/DevBoxClientTests.cs

```diff
@@ -0,0 +1,415 @@
+                }
+
+                string poolName = poolNameJson.ToString();
+                Assert.AreEqual(poolName, TestEnvironment.PoolName);
+            }
+
+            Assert.AreEqual(1, numberOfReturnedPools);
+        }
+
+        [RecordedTest]
+        public async Task GetSchedulesSucceeds()
+        {
+            var numberOfReturnedSchedules = 0;
+            await foreach (BinaryData scheduleData in _devBoxesClient.GetSchedulesAsync(TestEnvironment.ProjectName, TestEnvironment.PoolName))
+            {
+                numberOfReturnedSchedules++;
+                JsonElement getSchedulesResponseData = JsonDocument.Parse(scheduleData.ToStream()).RootElement;
+
+                if (!getSchedulesResponseData.TryGetProperty("name", out var poolNameJson))
+                {
+                    Assert.Fail("The JSON response received from the service does not include the necessary property.");
+                }
+
+                string poolName = poolNameJson.ToString();
+                Assert.AreEqual("default", poolName);
+            }
+
+            Assert.AreEqual(1, numberOfReturnedSchedules);
+        }
+
+        [RecordedTest]
+        public async Task GetScheduleSucceeds()
+        {
+            Response getScheduleResponse = await _devBoxesClient.GetScheduleAsync(
+                TestEnvironment.ProjectName,
+                TestEnvironment.PoolName,
+                "default");
+
+            JsonElement getScheduleData = JsonDocument.Parse(getScheduleResponse.ContentStream).RootElement;
+
+            if (!getScheduleData.TryGetProperty("name", out var scheduleNameJson))
+            {
+                Assert.Fail("The JSON response received from the service does not include the necessary property.");
+            }
+
+            string scheduleName = scheduleNameJson.ToString();
+            Assert.AreEqual("default", scheduleName);
+        }
+
+        [RecordedTest]
+        public async Task GetActionsSucceeds()
+        {
+            var numberOfReturnedActions = 0;
+            await foreach (BinaryData actionsData in _devBoxesClient.GetActionsAsync(TestEnvironment.ProjectName, DevBoxName))
+            {
+                numberOfReturnedActions++;
+                JsonElement getActionsResponseData = JsonDocument.Parse(actionsData.ToStream()).RootElement;
+
+                if (!getActionsResponseData.TryGetProperty("name", out var actionNameJson))
+                {
+                    Assert.Fail("The JSON response received from the service does not include the necessary property.");
+                }
+
+                string actionName = actionNameJson.ToString();
+                Assert.AreEqual("schedule-default", actionName);
+            }
+
+            Assert.AreEqual(1, numberOfReturnedActions);
+        }
+
+        [RecordedTest]
+        public async Task GetActionSucceeds()
+        {
+            Response getActionResponse = await _devBoxesClient.GetActionAsync(
```

Comment:
Doesn't this test (and the skip and delay tests as well) need to run `SetUpDevBoxAsync` as a prerequisite? In particular because, if you try to perform skip then delay on the same dev box, I believe that would fail. After a skip, the dev box _should_ have no actions that can be delayed for some amount of time.

Reply:
Addressed here https://github.com/Azure/azure-sdk-for-net/pull/36245/commits/c6a5c6520cb385783cf8f31f5b4b60fba520aa31